core/src/consensus/tendermint/mod.rs

```diff
@@ -435,6 +440,13 @@ impl Tendermint {
                 ctrace!(ENGINE, "Transition to {:?} triggered.", step);
                 self.to_step(step);
             }
+        } else if vote_step.step == Step::Precommit
+            && self.height() - 1 == vote_step.height
```

Comment:
Is it okay with the genesis block?
The hight of the genesis block is 0.

Reply:
`self.height()` has positive value always.

Reply:
You are right. it's always a positive value.
I checked that it's the height to vote, not the current height.